pkg/scaffolding/scaffold.go

```diff
@@ -0,0 +1,87 @@
+package scaffolding
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"knative.dev/func/pkg/filesystem"
+)
+
+// Write scaffolding to a given path
+//
+// Scaffolding is a language-level operation which first detects the method
+// signature used by the function's source code and then writes the
+// appropriate scaffolding.
+//
+// NOTE: Scaffoding is not per-template, because a template is merely an
+// example starting point for a Function implementation and should have no
+// bearing on the shape that function can eventually take.  The language,
+// and optionally invocation hint (For cloudevents) are used for this.  For
+// example, there can be multiple templates which exemplify a given method
+// signature, and the implementation can be switched at any time by the author.
+// Language, by contrast, is fixed at time of initialization.
+//
+//	out:     the path to output scaffolding
+//	src:      the path to the source code to scaffold
+//	runtime: the expected runtime of the target source code "go", "node" etc.
+//	invoke:  the optional invocatin hint (default "http")
+//	fs:      filesytem which contains scaffolding at '[runtime]/scaffolding'
+//	         (exclusive with 'repo')
+func Write(out, src, runtime, invoke string, fs filesystem.Filesystem) (err error) {
+
+	// detect the signature of the source code in the given location, presuming
+	// a runtime and invocation hint (default "http")
+	s, err := detectSignature(src, runtime, invoke)
+	if err != nil {
+		return err
+	}
+
+	// Path in the filesystem at which scaffolding is expected to exist
+	d := fmt.Sprintf("%v/scaffolding/%v", runtime, s.String()) // fs uses / on all OSs
+	if _, err := fs.Stat(d); err != nil {
+		return fmt.Errorf("no scaffolding found for '%v' signature '%v'. %v.  Note that this may require the filesystem be regenerated (try make first).", runtime, s, err)
```

Comment:
The last part of this error seems to be targeted at the `func` developer instead of a functions developer. I am a bit confused by it.